Runtime/Tracking/Collision/CollisionNotifier.cs

```diff
@@ -80,7 +80,7 @@ public bool Equals(EventData other)
                     return true;
                 }
 
-                return Equals(ColliderData.GetContainingTransform(), other.ColliderData.GetContainingTransform()) && ((ForwardSource == null && other.ForwardSource == null) || Equals(ForwardSource, other.ForwardSource));
+                return Equals(ColliderData, other.ColliderData) && ((ForwardSource == null && other.ForwardSource == null) || Equals(ForwardSource, other.ForwardSource));
             }
 
             /// <inheritdoc />

```

Comment:
```suggestion
            /// <inheritdoc />
            public override bool Equals(object obj)
            {
                if (ReferenceEquals(null, obj))
                {
                    return false;
                }

                if (ReferenceEquals(this, obj))
                {
                    return true;
                }

                return obj is EventData other && Equals(other);
            }
```
Note the last line: We want to allow subclassing this and thus equality needs to be overridable properly, which this handles.

Reply:
are you saying here replace
```
public override bool Equals(object obj)
{
    if (ReferenceEquals(null, obj))
    {
        return false;
    }

    if (ReferenceEquals(this, obj))
    {
        return true;
    }

    if (obj.GetType() != GetType())
    {
        return false;
    }

    return Equals((EventData)obj);
}
```

with

```
public override bool Equals(object obj)
{
  if (ReferenceEquals(null, obj))
  {
    return false;
  }
  if (ReferenceEquals(this, obj))
  {
    return true;
  }
  return obj is EventData other && Equals(other);
}
```

Reply:
Yes :) (Newlines or not to separate the `if`s I'm fine with 😄)

Reply:
The newlines were removed due to some bad copy/pasting :)